fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1307,14 +1363,59 @@ ACTOR Future<Void> dataDistributionRelocator(DDQueueData* self,
 
 						bestTeams.emplace_back(bestTeam.first.get(), bestTeam.second);
 					}
+
+					// update physicalShardIDCandidate
+					if (CLIENT_KNOBS->SHARD_ENCODE_LOCATION_METADATA && tciIndex==0 && rd.dataMoveIdSetDelayed) { // && rd.dataMoveIdSetDelayed && foundTeams
+						ASSERT(foundTeams);
```

Comment:
Why can we assert foundTeams here? Is it guaranteed that getTeam() will return a healthy team for the primary?

Reply:
If foundTeam==false, the loop breaks and will never reach this part.